### ISSUE TITLE
Harmonise users on drupal 7 and 8 sites

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/ABLE#53]
In preparation for switching to the drupal 8 sites, need to ensure that users are consistent between the drupal 7 site (www.butterfly-monitoring.net/mydata) and live drupal 8 site on pantheon (http://live-butterfly-monitoring.pantheonsite.io/).

Any conflicts to be reviewed by me

**Comment by DavidRoy** [comment]
@Gary-van-Breda this is becoming urgent as we have users beginning to register after workshops we are running

**Comment by Gary-van-Breda** [comment]
There are 2 users on the new system which are not on the old, and 9 active on the old site that are not on the new. At the moment the IDs don't overlap.
There are 8 instances of the sites being allocated to one of these new users, and 1 as a country manager.
The incremental migration path is not working.
I may have to create the new users manually, but this means I would have to change the 9 location assignments manually as well. With this small number thats not an issue, but if we continue to get new users it may become so. We should aim to transfer to the new site asap, and I will need to do a final user check at the point of transfer.
I'm just going to check if I can use a CSV [export/]import to do this, and set the CMS user IDs at the same time to the values on the old site: this will make our lives easier.

**Comment by DavidRoy** [comment]
Yes, please switch over asap to minimise further work to synchronise

**Comment by Gary-van-Breda** [comment]
I've activated the pre-installed views_data_export module on the original site, set up a view/file export to dump the users details (not passwords), accessible only by admins.

**Comment by Gary-van-Breda** [comment]
I have added user_csv_import contrib module, currently on Dev only, to test the importing of users.

**Comment by Gary-van-Breda** [comment]
I've uploaded the new users into the Dev, Test and Live environments.
A Modified user_csv_import allowed us to keep the same CMS user IDs.
The 11 users should all reset their passwords when they come to use the new system.
I have set the roles as best I can, but there are differences in the roles between the old and new sites, so these should be reviewed.
I have subsequently switched off the user_csv_import module on test and live.
